api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TXCXCACohortQueries.java

```diff
@@ -374,4 +369,762 @@ public CohortDefinition getFinalComposition(
+            + "                                         FROM       patient p "
+            + "                                                        INNER JOIN encounter e "
+            + "                                                                   ON         e.patient_id = p.patient_id "
+            + "                                                        INNER JOIN obs o "
+            + "                                                                   ON         o.encounter_id =e.encounter_id "
+            + "                                         WHERE      p.voided = 0 "
+            + "                                           AND        e.voided = 0 "
+            + "                                           AND        o.voided = 0 "
+            + "                                           AND        e.encounter_datetime BETWEEN :startDate AND        :endDate "
+            + "                                           AND        e.location_id = :location "
+            + "                                           AND        e.encounter_type = ${28} "
+            + "                                           AND        o.concept_id = ${2094} "
+            + "                                           AND        o.value_coded IN (${703}, "
+            + "                                                                        ${2093}, "
+            + "                                                                        ${664}) "
+            + "                                         GROUP BY   p.patient_id) AS last_result "
+            + "                                     ON         last_result.patient_id = p.patient_id "
+            + "                          WHERE      p.voided = 0 "
+            + "                            AND        e.voided = 0 "
+            + "                            AND        o.voided = 0 "
+            + "                            AND        e.encounter_type = ${28} "
+            + "                            AND        e.encounter_datetime = last_result.encounter_date "
+            + "                            AND        e.location_id = :location "
+            + "                            AND        o.concept_id = ${2094} "
+            + "                            AND        o.value_coded = ${703} "
+            + "                          GROUP BY   p.patient_id ) positive_via "
+            + "           WHERE      p.voided = 0 "
+            + "             AND        o.voided = 0 "
+            + "             AND        e.encounter_type = ${28} "
+            + "             AND        e.location_id = :location "
+            + "             AND        o.concept_id = ${2149} "
+            + "             AND        o.value_coded = ${23972} "
+            + "             AND        o.obs_datetime BETWEEN positive_via.last_positive_encounter AND        :endDate "
```

Comment:
This is not aligned with the requirement, there is a criteria missing **(or Qual foi o tratamento/avaliação no HdR = "Termocolagulação Feita.)**

Reply:
criteria "Qual foi o tratamento/avaliação no HdR" = "Termocolagulação Feita" is the one already done and the missing one is  Tratamento Feito = Termoablação, which is now done